roaster-selenide/roaster-selenide-api/src/main/java/hu/icellmobilsoft/roaster/selenide/SelenideConfigExtension.java

```diff
@@ -0,0 +1,96 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.roaster.selenide;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.Extension;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.weld.environment.se.events.ContainerInitialized;
+
+import com.codeborne.selenide.Configuration;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.WebDriverRunner;
+
+import hu.icellmobilsoft.coffee.se.logging.Logger;
+import hu.icellmobilsoft.roaster.selenide.config.SelenideConfig;
+import hu.icellmobilsoft.roaster.selenide.drivers.ChromeDeviceModeDriverProvider;
+import hu.icellmobilsoft.roaster.selenide.drivers.CustomIEDriverProvider;
+import hu.icellmobilsoft.roaster.selenide.enums.Browser;
+
+/**
+ * Selenide Configuration extension.
+ *
+ * @author speter555
+ * @since 0.2.0
+ */
+public class SelenideConfigExtension implements Extension {
+
+    /**
+     * Handle container Initialized event
+     *
+     * @param containerInitialized
+     *            instance
+     */
+    public void observesContainerInitialized(@Observes @Initialized(ApplicationScoped.class) ContainerInitialized containerInitialized) {
```

Comment:
@rombow ezt szerettem volna még kérdezni, hogy ez biztos jó lesz így? Mi van ha másnak is kell azaz event hogy elkészült a konténer? Ezt akkor kitehetnénk egy közös helyre, és ott további eventökön keresztül juthatnánk el ide.

Pl ez bekerül egy common részbe, és ott lehet hozzáadni különböző eventeket, pl SelenideContainerInitialized event, RedisContainerInitialized event ... Csak hogy ne vigye el egy modul ezt az event kezelést.

Reply:
Egyébként ez a konfig állítás miert Extension-ben lett megoldva? Ahogy nézem ez az egész WebDriverRunner full static. Nem lenne jobban kezelhető ha ős test osztalyban van valahol ami meghivja ezt az initDriver()-t?

Ez az annotacio sztem itt nincs is ilyen formaban hatassal semmire: @Initialized(ApplicationScoped.class), **@Observes ContainerInitialized event** elég kellene legyen. Részemről maradhat Extension is, de az inkabb arra valo hogy valamit CDI közelbe hozzon, de kb nem is találok ezekben a driverekben olyat ami static metodust. És igy elég nehéz lesz később ezeket a kongifokat rugalmassá tenni ugy hogy itt épül fel a driver.

Reply:
Sajna a meglátásod nagyon jó, tényleg minden static, azért hogy system property-ként futtatásnál lehessen beállítani rajta majdhogynem mindent. Ezt az egészet azért nem akartam oda kiszervezni hogy valamelyest el legyen rejtve.
Nem tudtam hogy így simán az `@Observes ContainerInitialized containerInitialized` működik, megcsináltam így, jó is, szóval a többi megoldást vissza is vonom.

Reply:
Ugylatszik hogy tenyleg full static, es a teszteken levo beallitas rahatassal lenne a masik tesztekre, tehat kozponti beallitas kellene, ugy ahogy peldaul ez az osztaly is van.